app/controllers/actions/AffinityGroupIdentifierAction.scala

```diff
@@ -44,8 +44,11 @@ class AffinityGroupIdentifierAction[A] @Inject()(action: Action[A],
                                 action: Action[A]
                                ) = {
 
+    val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
+    val sessionId = hc.sessionId.map(_.value).getOrElse("No Session ID available")
```

Comment:
Could this method be put in its own object so it doesn't have to be repeated everywhere?